src/renderer/components/swap/Swap.tsx

```diff
@@ -390,6 +397,142 @@ export const Swap = ({
+      ),
+    [sourceAsset, targetAsset]
+  )
+
+  const sourceChainBalanceError = useMemo(
+    () =>
+      FP.pipe(
+        sequenceTOption(sourceAsset, walletBalances),
+        O.map(([sourceAsset, walletBalances]) => {
+          // If sourceAsset is the same as source chain
+          // we should subtract changeAmount
+          const shouldSubtractChangeAmount = eqAsset.equals(sourceAsset, getChainAsset(sourceAsset.chain))
+
+          const balanceMinusAmount = FP.pipe(
+            walletBalances,
+            A.findFirst((walletBalance) => eqAsset.equals(walletBalance.asset, sourceAsset)),
+            O.map((sourceBalance) =>
+              baseToAsset(sourceBalance.amount)
+                .amount()
+                .minus(shouldSubtractChangeAmount ? changeAmount : ZERO_BN)
+            ),
+            O.map(FP.flow(assetAmount, assetToBase))
+          )
+
+          return FP.pipe(
+            sequenceTOption(balanceMinusAmount, oFeesByChainHashMap),
+            O.map(([leftBalance, fees]) => {
+              const sourceChainAssetString = assetToString(getChainAsset(sourceAsset.chain))
+              const sourceChainFeeAmount = fees[sourceChainAssetString]?.amount() || ZERO_BN
+
+              return leftBalance.amount().minus(sourceChainFeeAmount)
+            }),
+            O.map((result) => result.isNegative()),
+            O.getOrElse(() => false)
+          )
+        }),
+        O.getOrElse(() => false)
+      ),
+    [changeAmount, sourceAsset, walletBalances, oFeesByChainHashMap]
+  )
+
+  const sourceChainErrorLabel = useMemo(() => {
+    if (!sourceChainBalanceError) {
+      return <></>
+    }
+    const sourceChainAsset = FP.pipe(
+      sourceAsset,
+      O.map(({ chain }) => getChainAsset(chain)),
+      O.map((asset) => asset.symbol),
+      O.getOrElse(() => '')
+    )
+
+    return (
+      <Styled.BalanceErrorLabel>
+        {intl.formatMessage({ id: 'swap.errors.amount.balanceShouldCoverChainFee' }, { asset: sourceChainAsset })}{' '}
+      </Styled.BalanceErrorLabel>
+    )
+  }, [sourceChainBalanceError, sourceAsset, intl])
+
+  const targetChainBalanceError = useMemo(
+    () =>
+      FP.pipe(
+        // Dont need to calculate in case of same chain.
+        // In this case we aclready included it in sourceChainFee
+        sequenceTOption(targetAsset, walletBalances, oFeesByChainHashMap, isDifferentCains),
+        O.chain(([targetAsset, walletBalances, feesByChainHashMap]) => {
+          return FP.pipe(
+            walletBalances,
+            A.findFirst((walletBalance) => eqAsset.equals(walletBalance.asset, targetAsset)),
+            O.map((targetBalance) => {
+              const targetChainAsset = getChainAsset(targetAsset.chain)
+              const feeAmount = feesByChainHashMap[assetToString(targetChainAsset)]?.amount() || ZERO_BN
+              return targetBalance.amount.amount().minus(feeAmount).isNegative()
```

Comment:
> OutboundTransactionFee is what the user is charged (in the asset)

^ https://github.com/thorchain/asgardex-electron/issues/640

An user won't be charged for target chain fees from its balances - it's charged in the target asset. To validate fee for target chain, we have to make sure that the expected amount of target is more than 3x target chain fee (see #640.